tests/functional/adapter/test_anonymous_usage_stats.py

```diff
@@ -0,0 +1,43 @@
+from dbt.tests.util import run_dbt_and_capture
+import pytest
+
+
+ANONYMOUS_USAGE_MESSAGE = """
+sys._xoptions['snowflake_partner_attribution'].append("dbtLabs_dbtPython")
+""".strip()
+
+
+MY_PYTHON_MODEL = """
+import pandas
+
+def model(dbt, session):
+    dbt.config(materialized='table')
+    data = [[1,2]] * 10
+    return pandas.DataFrame(data, columns=['test', 'test2'])
+"""
+
+
+class AnonymousUsageStatsBase:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"my_python_model.py": MY_PYTHON_MODEL}
+
+
+class TestAnonymousUsageStatsOn(AnonymousUsageStatsBase):
+    @pytest.fixture(scope="class")
+    def profiles_config_update(self):
```

Comment:
This was updated manually because the way to flip this setting on/off is slightly different in 1.7. The same was done for this fixture the class below (line 38).